frontend/containers/forms/uploader/component.tsx

```diff
@@ -0,0 +1,183 @@
+        message: (
+          <FormattedMessage
+            defaultMessage="File {fileName} is larger than {fileSize}MB"
+            id="v9GdgG"
+            values={{
+              fileName: file.name,
+              fileSize: bytesToMegabytes(maxSize),
+            }}
+          />
+        ) as any,
+      });
+    },
+    [maxSize, setError]
+  );
+
+  const uploadFiles = useCallback(
+    async (files: File[]) => {
+      if (!files.length) return;
+      setFilesLength(files.length);
+      setIsUploading(true);
+
+      const uploadedFiles: ProjectImageGallery[] = [];
+
+      await Promise.all(
+        files.map(async (file) => {
+          if (file.size > maxSize) {
+            setMaxSizeError(file);
+          } else {
+            try {
+              const { signed_id, filename, direct_upload } = await directUpload(file);
+              uploadedFiles.push({
+                id: signed_id,
+                file: signed_id,
+                cover: false,
+                src: direct_upload?.url || URL.createObjectURL(file),
+                title: filename,
+              });
+            } catch (error) {
+              setError('project_images_attributes' as Path<FormValues>, {
+                message: error?.message || (
+                  <FormattedMessage
+                    defaultMessage="Something went wrong with the file upload"
+                    id="qpDURb"
+                  />
+                ),
+              });
+              setIsUploading(false);
+            }
+          }
+        })
+      );
+
+      onUpload(uploadedFiles);
+      setIsUploading(false);
+      setFilesLength(0);
+    },
+    [maxSize, onUpload, setError, setMaxSizeError]
+  );
+
+  const onDropAccepted = async (files: File[]) => {
+    clearErrors('project_images_attributes');
+    uploadFiles(files);
+  };
+
+  const onDropRejected = async (fileRejections: FileRejection[]) => {
+    setIsUploading(false);
+    fileRejections.forEach((rejection) => {
+      rejection.errors.forEach((error) => {
+        switch (error.code) {
+          case 'file-too-large':
+            setMaxSizeError(rejection.file);
+            break;
+          default:
+            setError('project_images_attributes' as Path<FormValues>, { message: error.message });
+        }
+      });
+    });
+  };
+
+  const { getRootProps, getInputProps } = useDropzone({
+    accept: fileTypes,
+    disabled: disabled || !!isUploading,
+    maxSize,
+    maxFiles,
+    multiple: maxFiles > 1,
+    onDropAccepted,
+    onDropRejected,
+  });
+
+  useEffect(() => {
+    if (isUploading) clearErrors('project_images_attributes');
+  }, [clearErrors, isUploading]);
+
+  return (
+    <div className="h-full">
+      <div
+        className={classNames(
+          'h-full rounded-md border-2 border-dashed  border-beige py-10 px-4 text-center text-sm focus:outline-green-700',
+          {
+            'cursor-pointer': !disabled,
+            'opacity-50': disabled,
```

Comment:
For disabled inputs, we usually use 60%.
```suggestion
            'opacity-60': disabled,
```